### PR TITLE
Member-Only rails app

### PR DESCRIPTION
**Pull request by MahmoudBakr23**
This app is built to implement the authentication and authorization methods introduced by Devise gem.

### CODE REVIEW

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello:wave: @MahmoudBakr23, @vmwhoami. Good start on this :+1: A few changes are required:**

:point_right: You have used the devise _authenticate\_member!_ helper method to restrict access to post's new and create actions :+1:. Now, you should create a custom method in Application Controller that mimics the _authenticate\_member!_ and use this method instead. The method should verify that a member is signed in and if not, redirect to sign in page.



:point_right: Extract the logic to display author name if a member is signed in into a helper method in ApplicationController and call the method in your posts index view.

:point_right: Sign in and Sign out links should be available in any view of the app. Place the links in Application Layout.




:point_right: Apart from the devise registration feature, implement your own custom user sign up feature. Set up the controller actions (new and create), views and routes required for this to work.

:point_right: _**Optional:**_ To make application controller methods available in your views, you can declare them as helper methods like shown below:

```ruby
helper_method :signed_in?
def signed_in?
  !current_user.nil?
end
```

_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested:exclamation:

**Comment by MahmoudBakr23** [comment]
Hi, Thanks a lot for your review,
1- I've done the helper method (member_authenticate!) in the application_controller to restrict the (new, create) methods as well!
2- I've used the views/layouts/application.html.erb/ for the sign_up and sign_in links.
3- Your first comment (as I understood) was referring to create a similar helper method to (member_authenticate!) and use it instead!
4- The second comment was telling me that I have to build a custom helper method in the application_controller to extract the logic for displaying the post author's name if logged_in? or not! Point 7 in the project tasks says otherwise!
5- Your last comment also doesn't exist in the project tasks.
I hope to kindly consider my comments and if there's anything I misunderstood, please let me know and I'll be more than happy to do what's required for me to do.
Thank you!

Re_  MahmoudBakr23_Members-Only  Member-Only rails app (#1) - mbakr6821@gmail com - Gmail - Google Chrome 9_24_2020 11_05_19 PM (2)

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <% if member_signed_in? %>
+        <td><strong> <%= post.member.name %> </strong></td><br />
+        <td><small> <%= post.member.username %> </small></td><br />
+        <td><%= post.post %></td>
+        <td><%= link_to 'Show', post %></td>
+        <td><%= link_to 'Edit', edit_post_path(post) %></td>
+        <td><%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% else %>
+        <td><%= post.post %></td>
+        <% end %>
```
Comment (lekegitrepo) [inline app/views/posts/index.html.erb]: - **_There should be absolutely NO logic in views._**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
### Hi @MahmoudBakr23 and @vmwhoami :clap: :clap:
#### Good job so far :+1: you have done a good job and the project looks good, however, you need to extract the logic from the view as mentioned in the last review.

**_Kindly, fix this and submit for another review._**

#### _Reviewed by Temitayo Adeleke._

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Well done, thank you for making the required changes by the previous TSE's

Great job guys, now you can merge your milestone is approved

## Happy Coding :tophat:

**Pull request merged**